FirebaseRemoteConfig/Tests/Sample/RemoteConfigSampleApp/ViewController.m

```diff
@@ -0,0 +1,501 @@
+  } else {
+    [[FRCLog sharedInstance] logToConsole:@"Nothing to set for defaults."];
+  }
+}
+
+- (IBAction)onClearLogsButtonPressed:(id)sender {
+  self.mainTextView.text = @"";
+}
+
+- (void)printDefaultConfigs {
+  NSMutableString *output = [[NSMutableString alloc] init];
+  [output appendString:@"\n-------Default config------\n"];
+  NSArray<NSString *> *result = [self.RCInstances[self.currentNamespace][self.FIRAppName]
+      allKeysFromSource:FIRRemoteConfigSourceDefault];
+  if (result) {
+    NSString *stringPerNs = @"";
+    for (NSString *key in result) {
+      FIRRemoteConfigValue *value =
+          [self.RCInstances[self.currentNamespace][self.FIRAppName] defaultValueForKey:key];
+      stringPerNs = [NSString stringWithFormat:@"%@%@ : %@ : %@\n", stringPerNs,
+                                               self.currentNamespace, key, value.stringValue];
+    }
+    [output appendString:stringPerNs];
+  }
+  [[FRCLog sharedInstance] logToConsole:output];
+}
+
+- (IBAction)getValueButtonPressed:(id)sender {
+  [[FRCLog sharedInstance] logToConsole:[self.RCInstances[self.currentNamespace][self.FIRAppName]
+                                            configValueForKey:self.keyLabel.text]
+                                            .debugDescription];
+}
+
+- (IBAction)logValueButtonPressed:(id)sender {
+  [[FRCLog sharedInstance]
+      logToConsole:[NSString stringWithFormat:@"key: %@ logged", self.keyLabel.text]];
+}
+
+// add default variable button pressed
+- (IBAction)addButtonPressed:(id)sender {
+  [self addNewEntryToVariables:self.configDefaults isDefaults:YES];
+}
+
+- (IBAction)developerModeSwitched:(id)sender {
+  FIRRemoteConfigSettings *configSettings =
+      [[FIRRemoteConfigSettings alloc] initWithDeveloperModeEnabled:self.developerModeEnabled.isOn];
+  ((FIRRemoteConfig *)(self.RCInstances[self.currentNamespace][self.FIRAppName])).configSettings =
+      configSettings;
+  [[FRCLog sharedInstance]
+      logToConsole:[NSString
+                       stringWithFormat:@"Developer Mode Enabled: %d\n",
+                                        ((FIRRemoteConfig *)(self.RCInstances[self.currentNamespace]
+                                                                             [self.FIRAppName]))
+                                            .configSettings.isDeveloperModeEnabled]];
+}
+
+- (void)addNewEntryToVariables:(NSMutableDictionary *)variables isDefaults:(BOOL)isDefaults {
+  if ([self.keyLabel.text length]) {
+    variables[self.keyLabel.text] = self.valueLabel.text;
+
+    NSString *showText = @"custom variables ";
+    if (isDefaults) {
+      showText = @"config defaults";
+    }
+    [[FRCLog sharedInstance]
+        logToConsole:[NSString stringWithFormat:@"New %@ added %@ : %@\n", showText,
+                                                self.keyLabel.text, self.valueLabel.text]];
+
+    self.keyLabel.text = @"";
+    self.valueLabel.text = @"";
+  }
+}
+
+- (void)apply {
```

Comment:
Can we qualify the method name to something like applyConfigs or activateConfigs?

Reply:
Ack. Am migrating the existing (internal) app for this CL. Will have a cleanup CL and use this input then.